src/script/jvm_script.cpp

```diff
@@ -59,7 +61,19 @@ bool JvmScript::inherits_script(const Ref<Script>& p_script) const {
 Ref<Script> JvmScript::get_base_script() const {
     if (!is_valid() || kotlin_class->registered_supertypes.size() == 0) { return {}; }
     StringName parent_name = kotlin_class->registered_supertypes[0];
-    return JvmScriptManager::get_instance()->get_script_from_name(parent_name);
+
+    //TODO: Remove with 4.4 when NamedScript names are GUIDs
+    for (const Dictionary& item: ProjectSettings::get_singleton()->get_global_class_list()) {
+        String class_name = item["class"];
+        if (class_name != parent_name) {
+            continue;
+        }
+
+        String script_path = item["path"];
+        return ResourceLoader::load(script_path);
+    }
+
```

Comment:
this needs to be updated. This PR will be merged after 4.4 is released from us so the base branch of this PR can already be the 4.4 update branch imo and this can be fixed already.
After 4.4 is merged into  master, gtihub will change the target branch to master automatically anyways